app/plugin/jira/plugin.go

```diff
@@ -0,0 +1,41 @@
+package jira
+
+import (
+	"crypto/subtle"
+	"encoding/json"
+	"net/http"
+
+	"github.com/mattermost/platform/app/plugin"
+)
+
+type Plugin struct {
+	api           plugin.API
+	Configuration Configuration
+}
+
+func (p *Plugin) Initialize(api plugin.API) {
+	p.api = api
+	if err := api.LoadConfiguration(&p.Configuration); err != nil {
+		panic(err)
+	}
+	api.Router().HandleFunc("/webhook", p.handleWebhook).Methods("POST")
+}
+
+func (p *Plugin) handleWebhook(w http.ResponseWriter, r *http.Request) {
+	if p.Configuration.Secret == "" || p.Configuration.UserId == "" {
+		http.Error(w, "This plugin is not configured.", http.StatusForbidden)
+	} else if subtle.ConstantTimeCompare([]byte(r.URL.Query().Get("secret")), []byte(p.Configuration.Secret)) != 1 {
+		http.Error(w, "You must provide the configured secret.", http.StatusForbidden)
```

Comment:
not localized. these strings aren't user-facing, and i didn't want to add jira-related strings to our main i18n files. i'd prefer to wait until we figure out a good strategy that allows plugin-specific translations to be fully encapsulated by the plugins